src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,891 @@
+                2020, 2021 Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2021 Pablo Escobar <mail@rvrs.in>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "KtxImporter.h"
+
+#include <map>
+#include <Corrade/Containers/Array.h>
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/Containers/StaticArray.h>
+#include <Corrade/Containers/String.h>
+#include <Corrade/Containers/StringView.h>
+#include <Corrade/Utility/Algorithms.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/Endianness.h>
+#include <Corrade/Utility/EndiannessBatch.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Math/BoolVector.h>
+#include <Magnum/Math/Functions.h>
+#include <Magnum/Math/Swizzle.h>
+#include <Magnum/Math/Vector3.h>
+#include <Magnum/Math/Vector4.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/TextureData.h>
+#include "MagnumPlugins/KtxImporter/KtxHeader.h"
+
+namespace Magnum { namespace Trade {
+
+namespace {
+
+std::size_t imageLength(const Vector3i& size, PixelFormat format) {
+    return size.product()*pixelSize(format);
+}
+
+/** @todo Use CompressedPixelStorage::dataProperties for this */
+std::size_t imageLength(const Vector3i& size, CompressedPixelFormat format) {
+    const Vector3i blockSize = compressedBlockSize(format);
+    const Vector3i blockCount = (size + (blockSize - Vector3i{1})) / blockSize;
+    return blockCount.product()*compressedBlockDataSize(format);
+}
+
+template<std::size_t Size> struct TypeForSize {};
+template<> struct TypeForSize<1> { typedef UnsignedByte  Type; };
+template<> struct TypeForSize<2> { typedef UnsignedShort Type; };
+template<> struct TypeForSize<4> { typedef UnsignedInt   Type; };
+template<> struct TypeForSize<8> { typedef UnsignedLong  Type; };
+
+/** @todo Can we perform endian-swap together with the swizzle? Might get messy
+          and it'll be untested... */
```

Comment:
Sounds like a good idea for a utility directly in Magnum. Goes onto my TODO list :)